src/evaluate/evaluator/token_classification.py

```diff
@@ -0,0 +1,277 @@
+            random_state (`int`, *optional*, defaults to `None`):
+                The `random_state` value passed to `bootstrap` if `"bootstrap"` strategy is chosen. Useful for
+                debugging.
+            input_column (`str`, defaults to `"tokens"`):
+                the name of the column containing the tokens feature in the dataset specified by `data`.
+            label_column (`str`, defaults to `"label"`):
+                the name of the column containing the labels in the dataset specified by `data`.
+            label_mapping (`Dict`, *optional*, defaults to `None`):
+                We want to map class labels defined by the model in the pipeline to values consistent with those
+                defined in the `label_column` of the `data` dataset.
+            join_by (`str`, *optional*, defaults to `" "`):
+                This evaluator supports dataset whose input column is a list of words. This parameter specifies how to join
+                words to generate a string input. This is especially useful for languages that do not separate words by a space.
+
+        Return:
+            A `Dict`. The keys represent metric keys calculated for the `metric` spefied in function arguments. For the
+            `"simple"` strategy, the value is the metric score. For the `"bootstrap"` strategy, the value is a `Dict`
+            containing the score, the confidence interval and the standard error calculated for each metric key.
+
+        The dataset input and label columns are expected to be formatted as a list of words and a list of labels respectively, following [conll2003 dataset](https://huggingface.co/datasets/conll2003). Datasets whose inputs are single strings, and labels are a list of offset are not supported.
+
+        Examples:
+        ```python
+        >>> from evaluate import evaluator
+        >>> from datasets import load_dataset
+        >>> task_evaluator = evaluator("token-classification")
+        >>> data = load_dataset("conll2003", split="validation[:2]")
+        >>> results = task_evaluator.compute(
+        >>>     model_or_pipeline="elastic/distilbert-base-uncased-finetuned-conll03-english",
+        >>>     data=data,
+        >>>     metric="seqeval",
+        >>>     join_by=" ",
+        >>> )
+        ```
+
+        <Tip>
+
+        For example, the following dataset format is accepted by the evaluator:
+
+        ```python
+        dataset = Dataset.from_dict(
+            mapping={
+                "tokens": [["New", "York", "is", "a", "city", "and", "Felix", "a", "person", "."]],
+                "ner_tags": [[1, 2, 0, 0, 0, 0, 3, 0, 0, 0]],
+            },
+            features=Features({
+                "tokens": Sequence(feature=Value(dtype="string")),
+                "ner_tags": Sequence(feature=ClassLabel(names=["O", "B-LOC", "I-LOC", "B-PER", "I-PER"])),
+                }),
+        )
+        ```
```

Comment:
the inputs could also be ints with a label_mapping right? so i think we should add an examples or/and explain more explicitly in the docstring what is allowed.

Reply:
Do you have an example in mind?

Reply:
Nevermind, all good :)